packages/eas-cli/src/build/metadata.ts

```diff
@@ -69,38 +74,38 @@ export async function collectMetadata<T extends Platform>(
   return sanitizeMetadata(metadata);
 }
 
-async function resolveAppVersionAsync<T extends Platform>(
-  ctx: BuildContext<T>,
-  platformContext?: MetadataContext<T>
-): Promise<string | undefined> {
-  if (ctx.platform === Platform.IOS) {
-    const iosContext = platformContext as IosMetadataContext | undefined;
-    return await readShortVersionAsync(ctx.projectDir, ctx.exp, iosContext?.buildSettings ?? {});
-  } else {
-    return await readVersionNameAsync(ctx.projectDir, ctx.exp);
-  }
-}
-
-async function resolveAppBuildVersionAsync<T extends Platform>(
+async function resolveVersionsAsync<T extends Platform>(
   ctx: BuildContext<T>,
   platformContext?: MetadataContext<T>
```

Comment:
```suggestion
  platformContext: MetadataContext<T>
```
It seems it's always defined.